src/main/java/graphql/nadel/engine/SourceQueryTransformer.java

```diff
@@ -0,0 +1,202 @@
+package graphql.nadel.engine;
+
+import graphql.analysis.QueryTraversal;
+import graphql.analysis.QueryVisitor;
+import graphql.analysis.QueryVisitorFieldEnvironment;
+import graphql.analysis.QueryVisitorFragmentSpreadEnvironment;
+import graphql.analysis.QueryVisitorInlineFragmentEnvironment;
+import graphql.execution.ExecutionContext;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.FieldDefinition;
+import graphql.language.FragmentDefinition;
+import graphql.language.InlineFragment;
+import graphql.language.Node;
+import graphql.language.ObjectTypeDefinition;
+import graphql.language.OperationDefinition;
+import graphql.language.TypeName;
+import graphql.language.VariableReference;
+import graphql.nadel.dsl.FieldDefinitionWithTransformation;
+import graphql.nadel.dsl.ObjectTypeDefinitionWithTransformation;
+import graphql.nadel.dsl.TypeTransformation;
+import graphql.nadel.engine.transformation.CopyFieldTransformation;
+import graphql.nadel.engine.transformation.FieldMappingTransformation;
+import graphql.nadel.engine.transformation.FieldTransformation;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.GraphQLType;
+import graphql.util.TreeTransformerUtil;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static graphql.Assert.assertFalse;
+import static graphql.Assert.assertNotNull;
+import static graphql.Assert.assertShouldNeverHappen;
+import static graphql.Assert.assertTrue;
+import static graphql.language.OperationDefinition.newOperationDefinition;
+import static graphql.language.SelectionSet.newSelectionSet;
+import static graphql.language.TypeName.newTypeName;
+import static graphql.util.TreeTransformerUtil.changeNode;
+import static java.util.function.Function.identity;
+
+public class SourceQueryTransformer {
```

Comment:
I am not so sure about the term source query: we don't use source anywhere else. Maybe more like: `OverallToUnderlyingQueryTransformer`?